### PR TITLE
Refactor signboard

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fixes [#839](https://github.com/mapbox/navigation-sdks/issues/839)

### Changelog

```
<changelog>Refactor signboard and alter api.</changelog>
```

### CODE REVIEW

File: libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/internal/SvgUtil.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package com.mapbox.navigation.ui.utils.internal
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import com.caverock.androidsvg.RenderOptions
+import com.caverock.androidsvg.SVG
+import java.io.ByteArrayInputStream
+
+object SvgUtil {
```
Comment (cafesilencio) [inline libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/internal/SvgUtil.kt:9]: Are there unit tests for this?

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt [hunk 2]
```diff
@@ -366,7 +366,7 @@ class MapboxRouteLineApi(
     ) {
         ThreadController.getMainScopeAndRootJob().scope.launch {
             val state = findClosestRoute(target, mapboxMap, padding)
-            resultConsumer.accept(state)
```
Comment (cafesilencio) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt:369]: "accept" is consistent with the Java naming convention for the consumer interface. What was the motivation for changing the name to consume?

Comment (abhishek1508) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt:369]: `consume` sounded more in line with `Consumer`. 🤷 
Will put it back though.

**Review by cafesilencio — APPROVED** [review]

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/signboard/api/MapboxSignboardApi.kt [hunk 3]
```diff
@@ -33,71 +35,41 @@ class MapboxSignboardApi(
...
      * [BannerComponents] of type [BannerComponents.GUIDANCE_VIEW] and subType [BannerComponents.SIGNBOARD]
      * @param instructions object representing [BannerInstructions]
-     * @param callback informs about the state of signboard.
+     * @param consumer informs about the state of signboard.
      */
-    override fun generateSignboard(
+    fun generateSignboard(
         instructions: BannerInstructions,
-        callback: SignboardReadyCallback
+        consumer: MapboxNavigationConsumer<Expected<SignboardValue, SignboardError>>
```
Comment (Guardiola31337) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/signboard/api/MapboxSignboardApi.kt:42]: We're using now the generic callback `MapboxNavigationConsumer`, should we update other features that still use regular callbacks for consistency @mapbox/navigation-android?

Comment (abhishek1508) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/signboard/api/MapboxSignboardApi.kt:42]: Yes we should definitely. I will cut a ticket and add the features that needs to be updated. We can tackle it in a separate PR.
cc @Guardiola31337

Comment (Guardiola31337) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/signboard/api/MapboxSignboardApi.kt:42]: Yeap! I was only flagging. Thanks for cutting the ticket.

**Pull request merged**